src/passes/Asyncify.cpp

```diff
@@ -914,8 +959,12 @@ struct AsyncifyFlow : public Pass {
     auto* set = curr->dynCast<LocalSet>();
     if (set) {
       auto type = set->value->type;
+      auto* setterVal = set->value;
+      if (type.isMulti()) {
+        setterVal = builder->makeTupleExtract(setterVal, 0);
```

Comment:
I'm confused. What is this `0`? Is that the only index here (if so, why is it a tuple)? Or where are the others handled?

Reply:
We need to keep track of the tuple from the original `local.set`, but it's not valid to pass the tuple as a parameter to the function. We make the function call valid by passing just a single element of the tuple, and later we can reach through the tuple.extract to recover the entire original tuple.

It might be conceptually cleaner to just split up the tuple and pass each of its elements to the function separately, but that would require a new temp variable to hold the local and would therefore be much more complicated.

Reply:
Interesting, thanks.

One thing I'm concerned about is optimizations. We do run the optimizer before and after `AsyncifyLocals`, so if it sees only part of the tuple is in use, it might end up eliminating code for the rest of the tuple. Perhaps the optimizer doens't do stuff like that for tuples now, but this would inhibit us from doing it later.